src/main/scala/com/typesafe/conductr/client/ConductRController.scala

```diff
@@ -163,18 +169,29 @@ class ConductRController(uri: Uri, connectTimeout: Timeout)
   import ConductRController._
   import context.dispatcher
 
-  override def receive: Receive = {
-    case GetBundleInfoStream   => fetchBundleFlow(sender())
-    case request: LoadBundle   => loadBundle(request)
-    case request: StartBundle  => startBundle(request)
-    case request: StopBundle   => stopBundle(request)
-    case request: UnloadBundle => unloadBundle(request)
+  override def receive: Receive = service(uri)
+
+  def service(host: Uri): Receive = {
+    case GetBundleInfoStream    => fetchBundleFlow(sender(), host)
+    case request: LoadBundle    => loadBundle(request, host)
+    case request: StartBundle   => startBundle(request, host)
+    case request: StopBundle    => stopBundle(request, host)
+    case request: UnloadBundle  => unloadBundle(request, host)
+    case SetControlServer(host) => setControlServer(host)
+
   }
 
   protected def request(request: HttpRequest, connection: Flow[HttpRequest, HttpResponse, Future[Http.OutgoingConnection]]): Future[HttpResponse] =
     Source.single(request).via(connection).runWith(Sink.head)
 
-  private def loadBundle(loadBundle: LoadBundle): Unit = {
+  private def setControlServer(host: URI): Unit = {
+    context.become(service(Uri(host.toString)))
+
+    // TODO verify this is a valid connection?
+    sender() ! "Control Server URL has be set to " + host.toString
```

Comment:
Sorry for being pesky, but I really think that there's no need to send a reply. Remove these lines and also the use of  the ask pattern in `TypesafeConductR.setControlServer`.